portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,31 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  /**
+   * returns login status: either null if not logged in or a string with their email
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String json;
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      json = gson.toJson(userEmail);
+    } else {
+      json = gson.toJson(null);
```

Comment:
Does this return an empty string?

Reply:
So on the client, I logged resp.json() and it came back as null (which is why my check was is json was falsey)

Reply:
I wonder what is the value of json in the server -- there are few possibilities for why `resp.json()` is null:

- `gson.toJson(null)` returns an empty string, therefore the response is empty. `resp.json()` in the client does not see any json data to parse, so it returns null.
-  `gson.toJson(null)` returns "null" (as a string), so the response is "null". `resp.json()` in the client see this and parse it as null.
- `gson.toJson(null)` returns null (non-string), and the response is "null".

I feel like it is probably not a good idea to depends on nullability?

Reply:
So i did a 
System.out.println(json); (where json was just gson.toJson(null) and it printed null

What do you think would be a better way to do it? Should I not be passing null into gson.toJson?

Reply:
Since all you need is a falsey value, you could directly write an empty string in the response (instead of using gson.toJson?)?

IMO the null and "null" conversion is confusing and unnecessary.

Reply:
are you saying for json to an empty string?
![image](https://user-images.githubusercontent.com/48307028/83913956-ed073800-a735-11ea-914c-3ef7ed6f1c14.png)

With this in script:
![image](https://user-images.githubusercontent.com/48307028/83913636-55094e80-a735-11ea-934f-0dd4bc902019.png)
It prints:
![image](https://user-images.githubusercontent.com/48307028/83913673-64889780-a735-11ea-88ce-e0e8fc7d4453.png)
with the gson.toJson(null)

but when I change it to the first picture
`json="";`
I get
![image](https://user-images.githubusercontent.com/48307028/83913900-ce08a600-a735-11ea-8883-ae8704307cbe.png)

Do you think I should put "" in the gson.toJson?

Reply:
Ok, so `JSON.parse("")` fails but `JSON.parse("null")` returns null :)

I guess it is fine to use gson.toJson(null) then, let's add some comments about this behavior.